torchvision/csrc/ops/roi_align.h

```diff
@@ -30,6 +30,123 @@ at::Tensor _roi_align_backward(
     int64_t sampling_ratio,
     bool aligned);
 
+template <typename T>
```

Comment:
I just moved the common code in the header, and added some comments / removed useless parameters. The rest of the code is unchanged.

Maybe there's a better place to put common code?

Reply:
The header files located at the root of `/ops/` folder contain the "public" methods for C++ API while the cpp files contain the code necessary for calling the dispatcher and registering the operators. They are not meant to contain code of the actual implementation of the method.

Up until now, there was very little need to share code between the CPU and CUDA implementations and as a result you won't find in torchvision's codebase an example that does what you want to do. One approach is to follow the pattern used at PyTorch and move the common code is separate files in the `/ops/cpu/` folder. See the [Loops.h](https://github.com/pytorch/pytorch/blob/c371542efc31b1abfe6f388042aa3ab0cef935f2/aten/src/ATen/native/cpu/Loops.h) file which is used both by [CPU kernels](https://github.com/pytorch/pytorch/blob/c371542efc31b1abfe6f388042aa3ab0cef935f2/aten/src/ATen/native/cpu/ComplexKernel.cpp) and [quantized](https://github.com/pytorch/pytorch/blob/c371542efc31b1abfe6f388042aa3ab0cef935f2/aten/src/ATen/native/quantized/cpu/qpool.cpp#L7).